Storage/SQL/SQLiteHistory.swift

```diff
@@ -451,46 +451,28 @@ extension SQLiteHistory: BrowserHistory {
             args = []
         }
 
-        let ungroupedSQL = [
-        "SELECT \(TableHistory).id AS historyID, \(TableHistory).url AS url, title, guid, domain_id, domain,",
-        "COALESCE(max(case \(TableVisits).is_local when 1 then \(TableVisits).date else 0 end), 0) AS localVisitDate,",
-        "COALESCE(max(case \(TableVisits).is_local when 0 then \(TableVisits).date else 0 end), 0) AS remoteVisitDate,",
-        "COALESCE(count(\(TableVisits).is_local), 0) AS visitCount",
-        "FROM \(TableHistory)",
-        "INNER JOIN \(TableDomains) ON \(TableDomains).id = \(TableHistory).domain_id",
-        "INNER JOIN \(TableVisits) ON \(TableVisits).siteID = \(TableHistory).id",
+        let sql = [
+        "SELECT",
+            "history.id AS historyID, history.url, title, guid, domain_id, domain,",
+            "COALESCE(MAX(CASE visits.is_local WHEN 1 THEN visits.date ELSE 0 END), 0) AS localVisitDate,",
+            "COALESCE(MAX(CASE visits.is_local WHEN 0 THEN visits.date ELSE 0 END), 0) AS remoteVisitDate,",
+            "COALESCE(COUNT(visits.is_local), 0) AS visitCount",
+            includeIcon ? ", iconID, iconURL, iconDate, iconType, iconWidth" : "",
+        "FROM",
+            "history",
+                "INNER JOIN domains ON domains.id = history.domain_id",
+                "INNER JOIN visits ON visits.siteID = history.id",
+                includeIcon ? "LEFT OUTER JOIN view_history_id_favicon ON view_history_id_favicon.id = history.id" : "",
         whereClause,
         "GROUP BY historyID",
-        ].joined(separator: " ")
-
-        let historySQL = [
-        "SELECT historyID, url, title, guid, domain_id, domain, visitCount,",
-        "max(localVisitDate) AS localVisitDate,",
-        "max(remoteVisitDate) AS remoteVisitDate",
-        "FROM (", ungroupedSQL, ")",
-        "WHERE (visitCount > 0)",    // Eliminate dead rows from coalescing.
-        "GROUP BY historyID",
-        "ORDER BY max(localVisitDate, remoteVisitDate) DESC",
+        "HAVING COUNT(visits.is_local) > 0",
```

Comment:
@rnewman The original query was doing this too except with a `WHERE (visitCount > 0)` in the outer-most `SELECT`. I'll look into all your feedback further when I get a chance, but in my dataset, I was getting the exact same results with my new query. I took the limit off the queries and output the results to CSV and did a `diff -u old.txt new.txt`, so I know I'm getting the exact same result with 12,224 rows which seems like a decently-large dataset.

Also, I ran the `EXPLAIN QUERY PLAN` on both queries and the new query only does 3 "scans" as opposed to the old query which does 5. To benchmark the query, I ran `.timer on` in sqlite and the new query was *slightly* faster. But, to be fair, the old query wasn't as slow for me as it was for you. So, I'd be curious to know how this patch performs with your data.